src/lib.rs

```diff
@@ -133,183 +114,114 @@ impl File {
     }
 
     /// Gets the wrapped file.
-    pub fn file(&self) -> &std::fs::File {
+    pub fn file(&self) -> &fs::File {
         &self.file
     }
 
     /// Gets the path of the wrapped file.
     pub fn path(&self) -> &Path {
         &self.path
     }
-
-    fn try_exec<R>(
-        &self,
-        op: impl FnOnce(&std::fs::File) -> std::io::Result<R>,
-        context: impl FnOnce(&path::Display) -> String,
-    ) -> Result<R, Error> {
-        match op(&self.file) {
-            Ok(result) => Ok(result),
-            Err(error) => Err(Error::new(error, context(&self.path.display()))),
-        }
-    }
```

Comment:
I replaced `try_exec` with just using `map_err` across the board

Reply:
The reason for `try_exec` was to let each function provide a custom error message based on the operation that failed - I would prefer not to throw that away

Reply:
I see! I've always thought of errors missing paths as the only issue with Rust's std FS errors, but they also don't include what operation they tried to perform.

Do you think it would make sense to re-introduce that custom error message by using an enum so that we can avoid allocating/formatting the error message?

Reply:
An enum would work, although since we need to clone the `PathBuf` already I don't think theres much advantage over just using `format!` directly